contracts/DomainRegister.sol

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.21;
+
+contract DomainRegister {
+    uint public  minimalPledge;
```

Comment:
Why not to use constant or immutable if you're not going to change the pledge? State var costs more gas.